app/src/main/java/nu/yona/app/api/service/ActivityMonitorService.java

```diff
@@ -106,15 +114,51 @@ private void restartReceiver()
 	public int onStartCommand(Intent intent, int flags, int startId)
```

Comment:
One last question: when is this triggered? When the device starts or when the app opens?

Reply:
`onStartCommand` is triggered when the ActivityMonitoringService starts by calling `startService` or `startForegroundService`. It happens in a couple below cases.
1. When user launches the application and passes pin code step. (starting already starting service will have no effect as it runs in same context. Any extras sent in second call to intent will get updated.)
2. When the app is removed from the recent applications. (Quitting from background)
3. Device is REBOOTED or Screen becomes active (service is stopped when screen goes black)

Reply:
OK, sounds good!